ddtrace/internal/_encoding.pyx

```diff
@@ -201,10 +371,13 @@ cdef class MsgpackEncoderBase(BufferedEncoder):
         """Return internal buffer."""
         return self.pk.buf + self._update_array_len()
 
+    cdef void * get_dd_origin_ref(self, object dd_origin):
```

Comment:
Is the type here `void *` since we don't know if it could be a string or an int in V03 and V05 respectively?

Reply:
That's right 🙁 . It's a bit of a hack that I'm not particularly happy with, but I couldn't think of another way of implementing this.